File: frontend/src/services/ProductMediaService.ts
/**
 * Product Media Service - API calls for image uploads
 */
import api from './api';
import type { ProductMediaResponse } from './interfaces/productTypes';

const BASE = '/products';

// ===== Thumbnail =====
export async function uploadThumbnail(
  productId: string,
  file: File,
): Promise<ProductMediaResponse> {
  const formData = new FormData();
  formData.append('file', file);

  return await api.post<ProductMediaResponse>(
    `${BASE}/${productId}/media/thumbnail`,
    formData,
    {
      headers: { 'Content-Type': 'multipart/form-data' },
    },
  );
}

export async function updateThumbnail(
  productId: string,
  file: File,
): Promise<ProductMediaResponse> {
  const formData = new FormData();
  formData.append('file', file);

  return await api.put<ProductMediaResponse>(
    `${BASE}/${productId}/media/thumbnail`,
    formData,
    {
      headers: { 'Content-Type': 'multipart/form-data' },
    },
  );
}

export async function deleteThumbnail(productId: string): Promise<ProductMediaResponse> {
  return await api.delete<ProductMediaResponse>(`${BASE}/${productId}/media/thumbnail`);
}

// ===== Gallery =====
export async function uploadGallery(
  productId: string,
  files: File[],
): Promise<ProductMediaResponse> {
  const formData = new FormData();
  files.forEach((file) => formData.append('files', file));

  return await api.post<ProductMediaResponse>(
    `${BASE}/${productId}/media/gallery`,
    formData,
    {
      headers: { 'Content-Type': 'multipart/form-data' },
    },
  );
}

export async function updateGalleryImage(
  productId: string,
  imageId: string,
  file: File,
): Promise<ProductMediaResponse> {
  const formData = new FormData();
  formData.append('file', file);

  return await api.put<ProductMediaResponse>(
    `${BASE}/${productId}/media/gallery/${imageId}`,
    formData,
    {
      headers: { 'Content-Type': 'multipart/form-data' },
    },
  );
}

export async function deleteGalleryImage(
  productId: string,
  imageId: string,
): Promise<ProductMediaResponse> {
  return await api.delete<ProductMediaResponse>(
    `${BASE}/${productId}/media/gallery/${imageId}`,
  );
}

File: backend/src/main/java/Bazaar/com/project/feature/Product/_ProductMedia/service/ProductMediaService.java
package Bazaar.com.project.feature.Product._ProductMedia.service;

import java.util.List;
import java.util.UUID;

import org.springframework.lang.NonNull;
import org.springframework.web.multipart.MultipartFile;

import Bazaar.com.project.feature.Product._ProductMedia.dto.ProductMediaResponse;

public interface ProductMediaService {
    ProductMediaResponse uploadThumbnail(@NonNull UUID productId, MultipartFile file);

    ProductMediaResponse uploadGallery(@NonNull UUID productId, List<MultipartFile> files);

    ProductMediaResponse updateThumbnail(@NonNull UUID productId, MultipartFile file);

    ProductMediaResponse updateGalleryImage(@NonNull UUID productId, @NonNull UUID imageId, MultipartFile file);

    ProductMediaResponse deleteThumbnail(@NonNull UUID productId);

    ProductMediaResponse deleteGalleryImage(@NonNull UUID productId, @NonNull UUID imageId);
}
